test/weets/post.spec.js

```diff
@@ -0,0 +1,96 @@
+
+  describe('Without an authenticated user', () => {
+    beforeAll(async () => {
+      weetCreationResponse = await httpRequest();
+    });
+
+    test('Responds with 401 status code', () => expect(weetCreationResponse.statusCode).toBe(401));
+
+    test('Responds with the expected error code', () =>
+      expect(weetCreationResponse.body.internal_code).toBe(AUTHORIZATION_ERROR));
+  });
+
+  describe('With an authorized user', () => {
+    let user = {};
+    let token = {};
+    let mock = {};
+
+    describe('When response from jokes api is succesful', () => {
+      beforeAll(async () => {
+        ({ user, token } = await authorizedUserWithToken());
+        mock = mockGeekJokesSuccessResponse();
+        weetCreationResponse = await httpRequest(token);
+      });
+      afterAll(() => truncateDatabase());
+      test('Responds with 201 status code', () => expect(weetCreationResponse.statusCode).toBe(201));
+
+      test('Responds with the expected schema', () =>
+        expect(weetCreationResponse.body).toMatchObject({
+          content: expect.any(String),
+          user_id: expect.any(Number)
+        }));
+
+      test('Creates a weet belonging to this user', () => expect(user.getWeets()).resolves.toHaveLength(1));
+
+      test('Calls mock', () => expect(mock.isDone()).toBe(true));
+    });
+
+    describe('With a failure response from jokes api', () => {
+      beforeAll(async () => {
+        ({ user, token } = await authorizedUserWithToken());
+        mock = mockGeekJokesFailureResponse();
+        weetCreationResponse = await httpRequest(token);
+      });
+      afterAll(() => truncateDatabase());
+
+      test('Responds with 502 status code', () => expect(weetCreationResponse.statusCode).toBe(502));
+
+      test('Responds with the expected error code', () =>
+        expect(weetCreationResponse.body.internal_code).toBe(EXTERNAL_SERVICE_ERROR));
+
+      test('Does not create a weet belonging to this user', () =>
+        expect(user.getWeets()).resolves.toHaveLength(0));
+
+      test('Calls mock', () => expect(mock.isDone()).toBe(true));
+    });
+
+    describe('When jokes api responds with a jokes with more than 140 characters', () => {
+      const largeJoke = chance.word({ length: 141 });
+
+      beforeAll(async () => {
+        ({ user, token } = await authorizedUserWithToken());
+        mock = mockGeekJokesSuccessResponse(largeJoke);
+        weetCreationResponse = await httpRequest(token);
+      });
+      afterAll(() => truncateDatabase());
+
+      test('Responds with 502 status code', () => expect(weetCreationResponse.statusCode).toBe(502));
+
+      test('Responds with the expected error code', () =>
+        expect(weetCreationResponse.body.internal_code).toBe(WEET_LENGTH_EXCEEDED));
+
+      test('Does not create a weet belonging to this user', () =>
+        expect(user.getWeets()).resolves.toHaveLength(0));
+
+      test('Calls mock', () => expect(mock.isDone()).toBe(true));
+    });
+  });
+});
```

Comment:
Great tests! 
I think we are missing the case when the token is from an nonexistent user. Correct me if I'm wrong

Reply:
You are right!